src/engraving/libmscore/edit.cpp

```diff
@@ -5496,12 +5504,35 @@ static void undoChangeRestVisibility(Rest* rest, bool visible)
     }
 }
 
+static void undoChangeOrnamentVisibility(Ornament* ornament, bool visible)
+{
+    ornament->undoChangeProperty(Pid::VISIBLE, visible);
+    Chord* cueNoteChord = ornament->cueNoteChord();
+    if (cueNoteChord) {
+        undoChangeNoteVisibility(cueNoteChord->upNote(), visible);
+    }
+    if (ornament->accidentalAbove()) {
+        ornament->accidentalAbove()->undoChangeProperty(Pid::VISIBLE, visible);
+    }
+    if (ornament->accidentalBelow()) {
+        ornament->accidentalBelow()->undoChangeProperty(Pid::VISIBLE, visible);
+    }
+}
+
 void Score::undoChangeVisible(EngravingItem* item, bool visible)
 {
     if (item->isNote()) {
         undoChangeNoteVisibility(toNote(item), visible);
     } else if (item->isRest()) {
         undoChangeRestVisibility(toRest(item), visible);
+    } else if (item->isOrnament()) {
```

Comment:
I expect to see new unit-tests for all these new cases

See changevisibility_tests.cpp

Reply:
@mike-spa please add the tests to your next PR. Thanks!